autoload/ale/cursor.vim

```diff
@@ -130,8 +130,14 @@ function! ale#cursor#ShowCursorDetail() abort
 
     if !empty(l:loc)
         let l:message = get(l:loc, 'detail', l:loc.text)
-
-        call ale#preview#Show(split(l:message, "\n"))
+        " In case options have been received, pass them down to the called
+        " method.
+        let options = get(a:000, 0, {})
+        if len(options) 
+            call ale#preview#Show(split(l:message, "\n"), options)
```

Comment:
Instead of passing on the same options this function received, create a new Dictionary which shows exactly which options you want to pass on explicitly.

Reply:
What would be the options I'm trying to pass down?

I thought about this and concerns were raised for me

1. Implicit coupling. Following your approach, when the called function changes, this would also need to change. And in case you were to forget about this detail, it would bug the code.
2. Assuming the first problem _isn't a problem_, how would I know which options need to be passed down, since it isn't explicit in `ale#preview#Show`'s signature?
3. Assuming this check could be be triggered for every CursorMove and there's no cost into passing the whole options dictionary as it comes, isn't it interesting to not overload this code with parsing and keep it slim, since it's tied to an AutoCmd? I'm thinking performance overhead there

Reply:
Don't pass on the same Dictionary arguments given to this function, as it's too hard to figure out which options are being given to the other one.